RockLib.Collections/NamedCollection.cs

```diff
@@ -6,203 +6,208 @@
 
 namespace RockLib.Collections
 {
-    /// <summary>
-    /// Represents a collection that can be retrieved by name.
-    /// </summary>
-    /// <typeparam name="T">The type of items in the collection.</typeparam>
-    public class NamedCollection<T> : IReadOnlyCollection<T>, IReadOnlyDictionary<string, T>
-    {
-        private const string _defaultDefaultName = "default";
-
-        private readonly IReadOnlyDictionary<string, T> _valuesByName;
-        private readonly Func<T, string> _dictionaryKeySelector;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="NamedCollection{T}"/> class
-        /// according to the specified <paramref name="getName"/> function.
-        /// </summary>
-        /// <param name="values">The values that make up the collection.</param>
-        /// <param name="getName">A function that gets the name of a value.</param>
-        /// <param name="stringComparer">
-        /// The equality comparer used to determine if names are equal. The default value is
-        /// <see cref="OrdinalIgnoreCase"/>.
-        /// </param>
-        /// <param name="defaultName">
-        /// The name that indicates an item should be used as the <see cref="DefaultValue"/>
-        /// for the <see cref="NamedCollection{T}"/>.
-        /// </param>
-        /// <param name="strict">
-        /// Whether to throw an exception if <paramref name="values"/> contains more than one
-        /// default value or more than one value with the same name. If <see langword="false"/>,
-        /// when there are duplicates, then the last value in the collection wins.
-        /// </param>
-        public NamedCollection(IEnumerable<T> values, Func<T, string> getName,
-            IEqualityComparer<string> stringComparer = null, string defaultName = _defaultDefaultName, bool strict = true)
-        {
-            if (values == null)
-                throw new ArgumentNullException(nameof(values));
-            if (getName == null)
-                throw new ArgumentNullException(nameof(getName));
-
-            StringComparer = stringComparer ?? OrdinalIgnoreCase;
-            DefaultName = !string.IsNullOrEmpty(defaultName) ? defaultName : _defaultDefaultName;
-
-            var valuesByName = new Dictionary<string, T>(StringComparer);
-            var hasDefaultValue = false;
-
-            foreach (var value in values)
+#pragma warning disable CA1033
```

Comment:
This one is weird to me.  Is this because this class isn't explicitly implementing methods that IReadOnlyDictionary is inheriting?  I guess it doesn't force you to implement them?

Reply:
I think this is similar to the other `#pragma` that I have. We should be writing code that adhers to [this analyzer](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1033), but since this is already done this way, I'd rather not risk breaking something.